temporalio/lib/temporalio/client/connection.rb

```diff
@@ -169,6 +169,9 @@ class HTTPConnectProxyOptions; end # rubocop:disable Lint/EmptyClass
       # @param lazy_connect [Boolean] If true, there is no connection until the first call is attempted or a worker
       #   is created with it. Clients from lazy connections cannot be used for workers if they have not performed a
       #   connection.
+      # @param around_connect [Proc, nil] If present, this proc accepts two values: options and a block. The block is a
+      #   must be yielded to only once with the options. The block does not return a meaningful value, nor should
```

Comment:
If I'm understanding correctly, we're the one yielding to the block and this isn't something in control of the callers
```suggestion
      #   will be yielded to once with the options. The block does not return a meaningful value, nor should
```

Reply:
The yielding is done by the one providing the `around_connect` proc parameter here, and we are saying they must yield to the block only once (not twice, not zero though technically zero is supported).